include/lsst/jointcal/AstrometryFit.h

```diff
@@ -52,13 +52,10 @@ class Associations;
  * CcdImage's. One does not need reference object and would then naturally not
  * have any Kind 2 terms.
  */
-class AstrometryFit {
+class AstrometryFit : public FitterBase {
 private:
-    Associations &_assoc;
-    std::string _whatToFit;
     bool _fittingDistortions, _fittingPos, _fittingRefrac, _fittingPM;
-    AstrometryModel *_astrometryModel;
-    int _LastNTrip;                   // last triplet count, used to speed up allocation
+    AstrometryModel &_astrometryModel;
```

Comment:
It's hard to know from just looking locally at the code whether `shared_ptr` will work; it's mostly a question of whether there would be cycles (unlike Python, `shared_ptr` doesn't do garbage collection).